packages/aws-library/src/aws_library/s3/_client.py

```diff
@@ -419,34 +422,20 @@ async def copy_objects_recursively(
         )
         if dst_metadata.size > 0:
             raise S3DestinationNotEmptyError(dst_prefix=dst_prefix)
-        try:
-
-            await limited_gather(
-                *[
-                    self.copy_object(
-                        bucket=bucket,
-                        src_object_key=s3_object.object_key,
-                        dst_object_key=s3_object.object_key.replace(
-                            src_prefix, dst_prefix
-                        ),
-                        bytes_transfered_cb=bytes_transfered_cb,
-                    )
-                    async for s3_object in self._list_all_objects(
-                        bucket=bucket, prefix=src_prefix
-                    )
-                ],
-                limit=_MAX_CONCURRENT_COPY,
-            )
-
-        except Exception:
-            # rollback changes
-            with log_catch(_logger, reraise=False), log_context(
-                _logger,
-                logging.ERROR,
-                msg="Unexpected error while copying files recursively, deleting partially copied files",
-            ):
-                await self.delete_objects_recursively(bucket=bucket, prefix=dst_prefix)
-            raise
+        await limited_gather(
+            *[
```

Comment:
MINOR: use a generator instead of a list i.e.  ` limited_gather(*(coro for coro in iter_coros))`  (see `(` vs `[`)

Reply:
yep, but still gather will anyway go through all the iterable, so probably the gain is really minimal